server/migrations/1660318202044_update_participant_status_infos.js

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable camelcase */
+const { dbClient, views } = require('../db');
+
+exports.shorthands = 'update-participant-status-infos';
+
+exports.up = async () => {
+  await dbClient.db.query(`CREATE OR REPLACE VIEW ${views.PARTICIPANTS_STATUS_INFOS} AS
+    SELECT p.id,
+     p.body,
+     p.search,
+     p.created_at,
+     p.updated_at,
+     jsonb_agg(ps.*) FILTER (WHERE ps.id IS NOT NULL) AS status_infos,
+     jsonb_agg(DISTINCT jsonb_build_object('id', rosstatus.id, 'data', rosstatus.data, 'status', rosstatus.status, 'site_id', rosstatus.site_id, 'rosSite', ros_site.*)) FILTER (WHERE rosstatus.id IS NOT NULL) AS ros_infos
```

Comment:
This is intense, I'd love an explanation

Reply:
Agreed

Reply:
This was tricky to build, but yes!
- this is the existing view `participants_status_infos` but with `ros_infos` added as a column
- I added the 2 joins to `ros_status` and `employer_sites` tables
- naturally I tried to add it like the `status_infos` column at first (doing `jsonb_agg` on the `ros_status` table, filtering out nulls)
- but there was a problem! Because it's an aggregate query, if I had 3 results from `participants_status` table and 1 from `ros_status` for a participant, I'd get 3 identical ros results
- each ros status was getting duplicated in the query for each participants status
- so I needed to do a `DISTINCT` on it to get rid of duplicates, but turns out that it doesn't know how to do a DISTINCT on json results
- that's what jsonb_build_object is for- turns it into something it can compare
- so I'm selecting distinct results from a json object that I've built

As for current...
I'm not 100% sure if we need a current filter. By the time the data makes its way to the FE, we're only looking at the ros status with the highest id. I don't know if there's any cases where the most recent ros status isn't valid but the prior status is.

Reply:
I understand, thanks!

I think I'd still urge for the current filter just for safeguarding. You never know in the future, or if someone makes manual database changes. We may as well take advantage of `is_current` where we have it!

Reply:
^ Agreed, for better or for worse it's a feature of these tables and if we have it we should be consistent about using it, leaving it out could cause another [HCAP-1265](https://freshworks.atlassian.net/browse/HCAP-1265)